flycheck.el

```diff
@@ -4875,9 +4875,15 @@ variable `flycheck-error-message-buffer'."
                                                flycheck-error-message-buffer
                                                'not-this-window)))
         (when (window-live-p result)
-          (with-current-buffer (window-buffer result)
-            (unless (derived-mode-p 'flycheck-error-message-mode)
-              (flycheck-error-message-mode))))))))
+          (let* ((maybe-popup-buffer (window-buffer result))
+                 (popup-buffer
+                  (if (string= (buffer-name maybe-popup-buffer)
+                               flycheck-error-message-buffer)
+                      maybe-popup-buffer
+                    (get-buffer flycheck-error-message-buffer))))
+            (with-current-buffer popup-buffer
+              (unless (derived-mode-p 'flycheck-error-message-mode)
+                (flycheck-error-message-mode)))))))))
```

Comment:
I think this is a bit convoluted.  Can we just (1) call `display-message-or-buffer` and (2) use `(get-buffer flycheck-error-message-buffer)` to get a handle to the message buffer, and change the mode there?

Reply:
I think you want `-when-let` here rather than `cl-letf*`

Reply:
oh yeah. done